src/borg/fuse.py

```diff
@@ -259,10 +260,17 @@ def _create_filesystem(self):
                     self.pending_archives[archive_inode] = archive.name
 
     def get_item(self, inode):
+        item = self._read_cache.get(inode)
+        if item:
+            return item        
         try:
-            return self._items[inode]
+            item = self._items[inode]
+            self._read_cache[inode] = item
+            return item
         except KeyError:
-            return self.cache.get(inode)
+            item = self.cache.get(inode)
+            self._read_cache[inode] = item 
+            return item
```

Comment:
I didn't read the full code right now, but that looks almost like putting a cache onto a cache (or something that could be a cache).

Reply:
Check comment for  self._items = {} :
        # Maps inode numbers to Item instances. This is used for synthetic inodes, i.e. file-system objects that are made up and are not contained in the archives. For example archive directories or intermediate directories not contained in archives.

File objects are never placed to self._items cache. So _read_cache is just a workaround for this problem, not permanent solution.